tests/net/mld/src/main.c

```diff
@@ -167,7 +167,12 @@ static void group_joined(struct net_mgmt_event_callback *cb,
 static void group_left(struct net_mgmt_event_callback *cb,
 			 uint32_t nm_event, struct net_if *iface)
 {
-	if (nm_event != NET_EVENT_IPV6_MCAST_LEAVE) {
+	uint32_t expected_event = _NET_EVENT_IPV6_BASE |
```

Comment:
Isn't this changing the meaning of the test? Previously a specific event was expected but now more are allowed.

Reply:
You are right, this is a bug. I have corrected this.
But there is still some problem in the test case. The k_yield fucnction works not as expected. I have changed it to k_sleep.
Thanks to review.